src/app/scripts/cac/search/cac-typeahead.js

```diff
@@ -54,10 +53,10 @@ CAC.Search.Typeahead = (function ($) {
         var typeaheadKey = $(event.currentTarget).data('typeahead-key') || defaultTypeaheadKey;
 
         if (dataset === 'currentlocation') {
-            self.events.trigger('cac:typeahead:selected', [typeaheadKey, suggestion]);
+            this.events.trigger('cac:typeahead:selected', [typeaheadKey, suggestion]);
         } else {
             CAC.Search.Geocoder.search(suggestion.text, suggestion.magicKey).then(function (location) {
-                self.events.trigger('cac:typeahead:selected', [typeaheadKey, location]);
+                this.events.trigger('cac:typeahead:selected', [typeaheadKey, location]);
```

Comment:
This (no pun intended) still throws an error -- the whole function needs to be reverted so it should be

```javascript
    function onTypeaheadSelected(event, suggestion, dataset) {
        var self = this;
        var typeaheadKey = $(event.currentTarget).data('typeahead-key') || defaultTypeaheadKey;

        if (dataset === 'currentlocation') {
            self.events.trigger('cac:typeahead:selected', [typeaheadKey, suggestion]);
        } else {
            CAC.Search.Geocoder.search(suggestion.text, suggestion.magicKey).then(function (location) {
                self.events.trigger('cac:typeahead:selected', [typeaheadKey, location]);
            });
        }
    }
```